### PR TITLE
Thin whitespaces must have their own span

### LINKED ISSUE #16224: Whitespace doesn't always belong to the right chunk when in marked content
Attach (recommended) or Link to PDF file here:
[whitespace_mc.pdf](https://github.com/mozilla/pdf.js/files/11091259/whitespace_mc.pdf)

Configuration:
- Web browser and its version: Firefox nightly
- Operating system and its version: Windows 11 

There's a whitespace (between `<` and `4`) which should have its own span because it's a marked content but it finally belongs to the chunk containing `4...`.

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: src/core/evaluator.js [hunk 1]
```diff
@@ -3305,6 +3326,7 @@ class PartialEvaluator {
             break;
           case OPS.endMarkedContent:
             flushTextContentItem();
+
```
Comment (Snuffleupagus) [inline src/core/evaluator.js:3329]: Nit: Unrelated change.

**Review by Snuffleupagus — APPROVED** [review]
r=me, with passing tests; thank you!

**Comment by calixteman** [comment]
There are few differences: in general it's an improvement.
When it isn't an improvement, it happens with scanned document: I compared with acrobat and it's almost the same.
